api/v1/inference_server.proto

```diff
@@ -37,16 +38,20 @@ message CreateChatCompletionRequest {
       string data = 1;
       string format = 2;
     }
+    message AudioUrl  {
+      string url = 1;
+    }
     message Content {
-      ContentType type = 1;
+      string type = 1;
       string text = 2;
       ImageUrl image_url = 3;
       InputAudio input_audio = 4;
+      AudioUrl audio_url = 5;
     }
     // Contents is different from openai API. In openAI API, the type of content can be string or array for User mesasges.
-    repeated Content contents = 5;
-    // Deprecated: use contents instead.
-    string content = 1;
+    repeated Content content = 5;
+    // Deprecated: use content instead.
+    string contents = 1;
```

Comment:
I see. We need to be careful on what will be broken by this. 

The OpenAI API reference has `string or array`, so this can break the compatibility with OpenAI as well. Keeping the `contents` field doesn't help as clients won't use the field.

We then also need to consider the backward compatibility between `server` and `engine`. If we don't keep the backward compatibility, it will be difficult to push this change.

We use this proto in the following places:

- unmarshall the request to extract the model ID
- communication protocol between the server and the engine
- request routing in the engine

I'm wondering if we can keep this as the `string` field and store the JSON string array there (or protocol message doesn't allow it?)

Reply:
I'm also wondering what's a better way to support `string or array` as specified in openai api. Union is not so compatible, how about `any`? I'm not sure how easy/hard for frontend(js) to handle it?

Reply:
Tried the several approach, but none of them worked. Let me try some more.

One approach would be to unmarshal a json without a proto, inspect the type content, and convert to an array. 

If this works, the remaining problem is the backward compatibility between the server and the engine.